src/components/actions/helpers.ts

```diff
@@ -44,7 +46,60 @@ export const swapOutAliasedActions = async (templateId: number, action: ActionIn
   if (altCondition !== undefined) aliasedAction.condition = altCondition
 
   // Override parameters
-  aliasedAction.parameter_queries = { ...aliasedAction.parameter_queries, ...overrideParams }
+  aliasedAction.parameter_queries = merge(aliasedAction.parameter_queries, overrideParams)
 
   return aliasedAction
 }
+
+interface ActionResult {
+  action: string // code
+  status: ActionQueueStatus
+  output: BasicObject | null
+  errorLog: string | null
+}
+
+interface ActionResultDisplayData {
+  type: 'DOCUMENT' | 'NOTIFICATION' | 'OTHER'
+  status: ActionQueueStatus
+  displayString: string
+  text?: string // for email string
+  fileId?: string // for generating url in front-end
+  errorLog: string | null // if Action failed
+}
+
+// Convert the full action output into a simplified format that can be easily
+// displayed by the front-end Preview module
+export const createDisplayData = (actionsOutput: ActionResult[]): ActionResultDisplayData[] => {
```

Comment:
feels like previewActionRoute is a good place for this one

Reply:
Yeah, sure: 1ef72368